### ISSUE TITLE
Corrupted light engine, macOS OpenGL.

### ISSUE THREAD

**Issue by Jakubk15** [Spottedleaf/Starlight#70]
Hello, when I launched Starlight first time on macOS Catalina computer, I saw some of blocks are not displaying properly. Only installed mods are Fabric API and latest version of Starlight.

I will attach example screenshots how it looks like.

![Zrzut ekranu 2021-06-26 o 11 18 29](https://user-images.githubusercontent.com/77227023/123508599-afdb5400-d670-11eb-8ba0-1ecb41cdd4f0.png)
![Zrzut ekranu 2021-06-26 o 11 18 16](https://user-images.githubusercontent.com/77227023/123508603-b4a00800-d670-11eb-876f-873a3a1c5700.png)

I bet this is caused because maybe OpenGL is not supported by Starlight. 

Computer specs:

Intel i5 4790R
Intel Iris Pro
8GB DDR3

**Comment by Kichura** [comment]
This is confirmed on phosphor, canvas and vanilla aswell but can sorta be resolved by cranking the smooth lighting to maxmium, but otherwise this is common with some 1.8.9 servers such as hypixel due to compatibility bugs with lighting not being respected properly. However on modern servers; this issue does not seem to be present whatsoever.

Smooth lighting set to disabled:
![no_sl](https://user-images.githubusercontent.com/68134602/123606386-22793a80-d7fd-11eb-9743-615cd2ee4a79.png)

Smooth lighting set to maxmium:
![sl](https://user-images.githubusercontent.com/68134602/123606420-2b6a0c00-d7fd-11eb-88b4-0048fc8e75f4.png)

Tested this via: Java 17-EA, Fabric 1.16.5, Windows Server 2019 with a Xeon CPU and GTX GPU.

**Comment by Spottedleaf** [comment]
Light data in older versions of the game propagated 0 to stairs/slabs, and it was the responsibility of the light read call to actually perform a recursive search of neighbours for correct values. Now the light engine in 1.14+ _Vanilla_ is capable of doing this kind of general logic in the propagator algorithm itself, so the light engine will expect this work to be done already. And on older servers it's just not done, so you see that stairs/slabs have light level 0.

Basically, not Starlight's problem. This will happen in Vanilla.

You can get a 100% clean fix by simply using an older client to connect to older servers. And no, hypixel doesn't run modern minecraft versions.
